homeassistant/components/august/__init__.py

```diff
@@ -75,138 +57,154 @@
     extra=vol.ALLOW_EXTRA,
 )
 
-AUGUST_COMPONENTS = ["camera", "binary_sensor", "lock"]
 
+async def async_request_validation(hass, config_entry, august_gateway):
+    """Request a new verification code from the user."""
 
-def request_configuration(hass, config, api, authenticator, token_refresh_lock):
-    """Request configuration steps from the user."""
+    _LOGGER.error("Access token is no longer valid.")
     configurator = hass.components.configurator
+    entry_id = config_entry.entry_id
 
-    def august_configuration_callback(data):
-        """Run when the configuration callback is called."""
-
-        result = authenticator.validate_verification_code(data.get("verification_code"))
+    async def async_august_configuration_validation_callback(data):
+        code = data.get(VERIFICATION_CODE_KEY)
+        result = await hass.async_add_executor_job(
+            august_gateway.authenticator.validate_verification_code, code
+        )
 
         if result == ValidationResult.INVALID_VERIFICATION_CODE:
-            configurator.notify_errors(
-                _CONFIGURING[DOMAIN], "Invalid verification code"
+            configurator.async_notify_errors(
+                _CONFIGURING[entry_id],
+                "Invalid verification code, please make sure you are using the latest code and try again.",
             )
         elif result == ValidationResult.VALIDATED:
-            setup_august(hass, config, api, authenticator, token_refresh_lock)
+            return await async_setup_august(hass, config_entry, august_gateway)
 
-    if DOMAIN not in _CONFIGURING:
-        authenticator.send_verification_code()
+        return False
+
+    if entry_id not in _CONFIGURING:
+        await hass.async_add_executor_job(
+            august_gateway.authenticator.send_verification_code
+        )
 
-    conf = config[DOMAIN]
-    username = conf.get(CONF_USERNAME)
-    login_method = conf.get(CONF_LOGIN_METHOD)
+    entry_data = config_entry.data
+    login_method = entry_data.get(CONF_LOGIN_METHOD)
+    username = entry_data.get(CONF_USERNAME)
 
-    _CONFIGURING[DOMAIN] = configurator.request_config(
-        NOTIFICATION_TITLE,
-        august_configuration_callback,
-        description=f"Please check your {login_method} ({username}) and enter the verification code below",
+    _CONFIGURING[entry_id] = configurator.async_request_config(
+        DEFAULT_NAME + " (" + username + ")",
```

Comment:
Use f-strings for string formatting
```suggestion
        f"{DEFAULT_NAME} ({username})",
```